clang/lib/CodeGen/CGObjCGNU.cpp

```diff
@@ -3993,7 +4009,9 @@ void CGObjCGNU::EmitThrowStmt(CodeGenFunction &CGF,
     ExceptionAsObject = CGF.ObjCEHValueStack.back();
     isRethrow = true;
   }
-  if (isRethrow && usesSEHExceptions) {
+  if (isRethrow &&
+      (usesSEHExceptions || (CGM.getTarget().getTriple().isOSCygMing() &&
+                             isRuntime(ObjCRuntime::GNUstep, 2)))) {
```

Comment:
Does it make sense to introduce a `usesCxxExceptions` variable?

Reply:
Yes, that makes the intent clearer.  I changed this.